webui/react/src/utils/task.ts

```diff
@@ -155,8 +155,7 @@ export const filterExperiments = (
         matchesUser<ExperimentItem>(experiment, users, filters.username) &&
         matchesState<ExperimentItem>(experiment, filters.states) &&
         matchesSearch<ExperimentItem>(experiment, search);
-    })
-    .slice(0, filters.limit);
```

Comment:
the local filter shouldn't be slicing here, `filters.limit` will be passed to the new list endpoints in the near future